### PR TITLE
MGMT-8461: Avoid panic in case nil is passed to GenerateError

### PR DESCRIPTION
**Pull request by ori-amizur**
# Assisted Pull Request

## Description
- Instead set error string "error is nil"
- In addition, fix a potential location that nil error was sent to  NewApiError

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @osherdp 
/cc @filanov 

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by osherdp — APPROVED** [review]
looks great!

File: internal/common/error_utils.go [hunk 1]
```diff
@@ -21,23 +21,23 @@ func (f NotFound) Error() string {
 }
 
 func GenerateError(id int32, err error) *models.Error {
+	var reason string
+	if err != nil {
+		reason = err.Error()
+	} else {
+		reason = "error is nil"
+	}
```
Comment (osherdp) [inline internal/common/error_utils.go:29]: sounds like a good idea :+1:

**Review by filanov — APPROVED** [review]

**Pull request merged**